src/app/pcasts/dao/listening_histories_dao.py

```diff
@@ -1,25 +1,38 @@
 from app.pcasts.dao import episodes_dao
 from . import *
 
-def create_or_update_listening_history(episode_id, user):
-  optional_episode = episodes_dao.get_episode(episode_id, user.id)
-  if optional_episode:
-    optional_listening_history = ListeningHistory.\
-      query.\
-      filter(
-          ListeningHistory.episode_id == episode_id,
-          ListeningHistory.user_id == user.id
-      ).first()
-
-    if optional_listening_history:
-      optional_listening_history.update_listening_time()
-      db_utils.db_session_commit()
-      return optional_listening_history
-
-    listening_history = ListeningHistory(episode_id=episode_id, user_id=user.id)
-    return db_utils.commit_model(listening_history)
-  else:
-    raise Exception('Invalid episode_id provided')
+def create_or_update_listening_histories(episode_update_info_map, user):
+  episode_ids = episode_update_info_map.keys()
```

Comment:
Not sure the rate at which we will have `real_duration` in the `episode_update_info_map` but if it is not very often, it might make more sense to get the episodes objects on demand, instead of doing a batch. Otherwise logic is really solid